Cannot access request header fields after connection is set

No idea how to reproduce, it was thrown at app launch.


### Actual Behavior
Crashed the app

```
02-08 13:17:11.645 11370-12377/com.mytest.app.android E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #4
          Process: com.mytest.app.android, PID: 11370
          java.lang.RuntimeException: An error occurred while executing doInBackground()
              at android.os.AsyncTask$3.done(AsyncTask.java:353)
              at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
              at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
              at java.util.concurrent.FutureTask.run(FutureTask.java:271)
              at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
              at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
              at java.lang.Thread.run(Thread.java:764)
           Caused by: java.lang.IllegalStateException: Cannot access request header fields after connection is set
              at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getRequestProperties(HttpURLConnectionImpl.java:232)
              at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getRequestProperties(DelegatingHttpsURLConnection.java:182)
              at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getRequestProperties(Unknown Source:0)
              at com.android.tools.profiler.support.network.httpurl.TrackedHttpURLConnection.getRequestProperties(TrackedHttpURLConnection.java:204)
              at com.android.tools.profiler.support.network.httpurl.TrackedHttpURLConnection.trackPreConnect(TrackedHttpURLConnection.java:63)
              at com.android.tools.profiler.support.network.httpurl.TrackedHttpURLConnection.getInputStream(TrackedHttpURLConnection.java:354)
              at com.android.tools.profiler.support.network.httpurl.HttpsURLConnection$.getInputStream(HttpsURLConnection$.java:261)
              at com.microsoft.codepush.react.CodePushUpdateManager.downloadPackage(CodePushUpdateManager.java:170)
              at com.microsoft.codepush.react.CodePushNativeModule$3.doInBackground(CodePushNativeModule.java:210)
              at com.microsoft.codepush.react.CodePushNativeModule$3.doInBackground(CodePushNativeModule.java:204)
              at android.os.AsyncTask$2.call(AsyncTask.java:333)
              at java.util.concurrent.FutureTask.run(FutureTask.java:266)
              at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
              at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
              at java.lang.Thread.run(Thread.java:764) 
```

### Environment

* react-native-code-push: **"5.2.1",**
* react-native: **"0.53.0",**
* Android version: **8.1.0**
* Does this reproduce on a debug build or release build? **Debug**
* Does this reproduce on a simulator, or only on a physical device? **Physical Device (Nexus 5X)**

--- a/android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateManager.java
+++ b/android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateManager.java
@@ -163,11 +163,11 @@ public class CodePushUpdateManager {
         try {
             URL downloadUrl = new URL(downloadUrlString);
             connection = (HttpURLConnection) (downloadUrl.openConnection());
+            bin = new BufferedInputStream(connection.getInputStream());
 
             long totalBytes = connection.getContentLength();
             long receivedBytes = 0;
 
-            bin = new BufferedInputStream(connection.getInputStream());
             File downloadFolder = new File(getCodePushPath());
             downloadFolder.mkdirs();
             downloadFile = new File(downloadFolder, CodePushConstants.DOWNLOAD_FILE_NAME);

